R/py_require.R

```diff
@@ -1,118 +1,183 @@
-#' Declare Python requirements
+#' Declare Python Requirements
 #'
-#' It allows you to specify the Python packages, and their versions, to use
-#' during your working session. It also allows to specify Python version
-#' requirements. It uses [uv](https://docs.astral.sh/uv/) to automatically
-#' resolves multiple version requirements of the same package (e.g.:
-#' 'numpy>=2.2.0', numpy==2.2.2'), as well as resolve multiple Python version
-#' requirements (e.g.: '>=3.10', '3.11').  `uv` will automatically download and
-#' install the resulting Python version and packages, so there is no need to
-#' take any steps prior to starting the Python session.
+#' `py_require()` allows you to declare Python requirements for the R session,
+#' including Python packages, any version constraints on those packages, and any
+#' version constraints on Python itself. Reticulate can then automatically
+#' create and use an ephemeral Python environment that satisfies all these
+#' requirements.
 #'
+#' Reticulate will only use an ephemeral environment if no other Python
+#' installation is found earlier in the [Order of
+#' Discovery](https://rstudio.github.io/reticulate/articles/versions.html#order-of-discovery).
+#' You can also force reticulate to use an ephemeral environment by setting
+#' `Sys.setenv(RETICULATE_USE_MANAGED_VENV = "yes")`.
 #'
-#' The virtual environment will not be initialized until the users attempts to
-#' interacts with Python for the first time during the session. Typically,
-#' that would be the first time `import()` is called.
+#' The ephemeral virtual environment is not created until the user interacts
+#' with Python for the first time in the R session. Typically, this occurs when
+#' `import()` is first called.
 #'
-#' If `uv` is not installed, `reticulate` will attempt to download and install
-#' a version of it in an isolated folder. This will allow you to get the
-#' advantages of `uv`, without modifying your computer's environment.
+#' If `py_require()` is called with new requirements after Reticulate has
```

Comment:
Maybe add `py_install()`: "if `py_require()`, or `py_install()`, is called..."?

Reply:
We currently emit a warning if `py_install()` is called on an ephemeral environment, I don't think we want to encourage this workflow.